scripts/api-extractor.ts

```diff
@@ -0,0 +1,286 @@
+    const projectFolder = resolvePath(__dirname, '..', packageDir);
+    const packagePath = resolvePath(__dirname, `../${packageDir}/package.json`);
+
+    const extractorConfig = ExtractorConfig.prepare({
+      configObject: {
+        mainEntryPointFilePath: resolvePath(
+          __dirname,
+          '../dist-types/packages/<unscopedPackageName>/src/index.d.ts',
+        ),
+        bundledPackages: [],
+
+        compiler: {
+          tsconfigFilePath: resolvePath(__dirname, '../tsconfig.json'),
+        },
+
+        apiReport: {
+          enabled: true,
+          reportFileName: 'api-report.md',
+          reportFolder: projectFolder,
+          reportTempFolder: resolvePath(outputDir, '<unscopedPackageName>'),
+        },
+
+        docModel: {
+          enabled: true,
+          apiJsonFilePath: resolvePath(
+            outputDir,
+            '<unscopedPackageName>.api.json',
+          ),
+        },
+
+        dtsRollup: {
+          enabled: false,
+        },
+
+        tsdocMetadata: {
+          enabled: false,
+        },
+
+        messages: {
+          compilerMessageReporting: {
+            default: {
+              logLevel: 'warning' as ExtractorLogLevel.Warning,
+              addToApiReportFile: true,
+            },
+          },
+          extractorMessageReporting: {
+            default: {
+              logLevel: 'warning' as ExtractorLogLevel.Warning,
+              addToApiReportFile: true,
+            },
+          },
+          tsdocMessageReporting: {
+            default: {
+              logLevel: 'warning' as ExtractorLogLevel.Warning,
+              addToApiReportFile: true,
+            },
+          },
+        },
+
+        newlineKind: 'lf',
+
+        projectFolder,
+      },
+      configObjectFullPath: projectFolder,
+      packageJsonFullPath: packagePath,
+    });
+
+    if (!compilerState) {
+      compilerState = CompilerState.create(extractorConfig, {
+        additionalEntryPoints: entryPoints,
+      });
+    }
+
+    // Message verbosity can't be configured, so just skip the check instead
+    (Extractor as any)._checkCompilerCompatibility = () => {};
+
+    // Invoke API Extractor
+    const extractorResult = Extractor.invoke(extractorConfig, {
+      localBuild: isLocalBuild,
+      showVerboseMessages: false,
+      showDiagnostics: false,
+      messageCallback(message) {
+        if (
+          message.text.includes(
+            'You have changed the public API signature for this project.',
+          )
+        ) {
+          console.log('');
+          console.log(
+            '*************************************************************************************',
+          );
+          console.log(
+            '* You have uncommitted changes to the public API of a package.                      *',
+          );
+          console.log(
+            '* To solve this, run `yarn build:api-reports` and commit all api-report.md changes. *',
```

Comment:
This is the feedback you get as a contributor in CI if you forget to commit api report changes